tutoraspects/templates/openedx-assets/queries/fact_enrollment_pii.sql

```diff
@@ -8,7 +8,10 @@ with
         from {{ DBT_PROFILE_TARGET_DATABASE }}.fact_enrollment_status enrollment
         inner join
             {{ ASPECTS_EVENT_SINK_DATABASE }}.course_names as course_names
-            on course_names.course_key = enrollment.course_key
+        -- Need to cast the course key to a string here otherwise the
+        -- course_names dictionary throws this:
+        -- Key type for complex key at position 0 does not match, expected String, found LowCardinality(String).
+        on course_names.course_key = enrollment.course_key::String
```

Comment:
Doesn't this affect DBT reports too?

Reply:
And custom reports that do the same join such as the `fact_*_engagement` ones?

Reply:
The `fact_*_engagement` charts are working as-is, but I can't guarantee what other custom reports might run into. That's likely a problem with a number of changes we've been making, though.